### PR TITLE
revert jdk target to 1.8

### PR DESCRIPTION
**Pull request by RBusarow**

### CODE REVIEW

**Review by steve-the-edwards — APPROVED** [review]

File: benchmarks/dungeon-benchmark/build.gradle.kts [hunk 1]
```diff
@@ -13,12 +15,12 @@ android {
   compileSdk = libsCatalog.version("compileSdk").toInt()
 
   compileOptions {
-    sourceCompatibility = JavaVersion.VERSION_17
-    targetCompatibility = JavaVersion.VERSION_17
+    sourceCompatibility = javaTargetVersion
```
Comment (rjrjr) [inline benchmarks/dungeon-benchmark/build.gradle.kts:18]: Do we really need to change `sourceCompatilibity` too? Will that, say, prevent us using language features that aren't actually supported?

**Review by rjrjr — APPROVED** [review]

File: benchmarks/dungeon-benchmark/build.gradle.kts (hunk 1 above)
Comment (RBusarow) [reply benchmarks/dungeon-benchmark/build.gradle.kts:18]: That's correct keeping them in sync will prevent us from inadvertently using Java language features that don't exist in the target environment.

File: build-logic/src/main/java/com/squareup/workflow1/buildsrc/internal/catalogs.kt [hunk 2]
```diff
@@ -44,3 +27,48 @@ fun DependencyHandler.kotlin(
   module: String,
   version: String? = null
 ): Any = "org.jetbrains.kotlin:kotlin-$module${version?.let { ":$version" } ?: ""}"
+
+/**
+ * the jdk used in packaging
```
Comment (rjrjr) [inline build-logic/src/main/java/com/squareup/workflow1/buildsrc/internal/catalogs.kt:32]: Isn't this redundant w/kgx?

Comment (RBusarow) [reply build-logic/src/main/java/com/squareup/workflow1/buildsrc/internal/catalogs.kt:32]: I'd thought it was, but sadly no.  The catalog entry/property names are unique to individual projects.

**Pull request merged**